### DISCUSSION TITLE
Saving authentication state - Flutter

### DISCUSSION THREAD

**Discussion by globemediaofficial** [supabase/supabase#9587] [Questions] [answered]
With Firebase, logging in a user saves the user to the device. Their authentication state is saved, and when they terminate and re-open the app, they are still logged in to the same account. That's automatic with Firebase, but Supabase doesn't seem to do it. When I run my Flutter app on any platform, I have to log in each time I open the app. How do I get around that? Why is this not automatic?

**Comment by mashood100** [comment]
you can save your own state in your local file storage by sharing preference package here is how:
initialised the sharePreferenceces in your main
 ``` dart
  
late SharedPreferences sharedPreferences;
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  sharedPreferences = await SharedPreferences.getInstance();
  runApp(const MyApp());
}
```

save state on login button as

```dart
await sharedPreferences.setBool('islogin', true);

```

check state when app start as 




```dart

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  
@override
  Widget build(BuildContext context) {
    bool islogin =    sharedPreferences .getBool('islogin');
    
        return MaterialApp(
        home: islogin == null ? const loginScreen() : const Homepage());
  }
}
```

**Reply by globemediaofficial** [reply]
That would work if I only needed to know if the user was authenticated at some point. However, I need to know that as well as access the database/storage, update the user, etc. To do this with Row Level Security the app needs the Supabase authentication saved, not just a boolean. The solution was the use the “supabase_flutter” package instead of “supabase”. With that package you can specify a secure storage and everything will be saved there automatically. I have no idea why there are 2 packages, both are updated and made by the Supabase team.

**Comment by globemediaofficial** [comment, marked as answer]
The solution was the use the “supabase_flutter” package instead of “supabase”. With that package you can specify a secure storage and everything will be saved there automatically. I have no idea why there are 2 packages, both are updated and made by the Supabase team.

**Reply by mashood100** [reply]
oh that's good I thought you only want to navigate to the home screen if the user is logged in

**Reply by dshukertjr** [reply]
Sorry for the confusion there. I have updated the readme, docs, and where ever possible that I can update to make it more clear to use supabase-flutter for flutter projects in the past few weeks. supabase package exists to be used in non-Flutter projects such as server side Dart.
